### ISSUE TITLE
Research pallet to to replace generic-asset

### ISSUE THREAD

**Issue by whalelephant** [subdarkdex/subdex-parachains#1]
Generic asset pallet has been remove from v2.0.0 release, [reaons here](https://github.com/paritytech/substrate/pull/7156)

There have been two proposals to replace pallet-generic-assets with: 
1. [pallet-assets](https://github.com/paritytech/substrate/blob/master/frame/assets/src/lib.rs)
2. [tokens](https://github.com/open-web3-stack/open-runtime-module-library/tree/master/tokens)

There is an active PR https://github.com/paritytech/substrate/pull/7156 to add functionalities to the first option.

We should decide on this as it is a blocker to both parachains (generic and subdex) development

**Comment by whalelephant** [comment]
I suggest we use pallet-assets as it is actively being developed and maintained. The PR above will have additional functionalities that the open-runtime-module-lib/token do not (yet?) have. Since the aim of this milestone is to provide reusable pallets for the ecosystem, this might makes maintaining the pallets produced easier as we do not have to worry about if other projects have the same dependency versions.

**Comment by fyang1024** [comment]
Yes, I think the asset pallet is more likely to to used for future cross-chain asset mapping

**Comment by iorveth** [comment]
I am concerned that pallet-assets lack of `mint/burn` functionality.  How xcmp transfers will be implemented than?
Suppose we can split this into few steps: 
1. Create new xcmp message to pass total issuance along with actual amount to be transferred. 
1. Issue new wrapped token/perform wrapped currencies transfers from/to root account each time, xcmp message transfer occurs.

Though this does not look clear and we need to send message back to ensure funds were moved to our dex account on another para-chain.  
If we go this way, i think it would even better to implement our own xcmp messages standard (invest, divest, exchange, etc.) to operate in fully remote way and distribute it as plugin.
 
What do you think guys?

**Comment by iorveth** [comment]
I am concerned that pallet-assets lack of `mint/burn` functionality.  How xcmp transfers will be implemented than?
Suppose we can split the workflow into few steps: 
1. Create new xcmp message to pass total issuance along with actual amount to be transferred. 
1. Issue new wrapped token/perform wrapped currencies transfers from/to root account each time, xcmp message transfer occurs.

Though this does not look clear and we need to send message back to ensure funds were moved to our dex account on another para-chain.  
If we go this way, i think it would even better to implement our own xcmp messages standard (invest, divest, exchange, etc.) to operate in fully remote way and distribute it as plugin.
 
What do you think guys? @whalelephant @fyang1024 @KStasi @herryho

**Comment by herryho** [comment]
I read the pallet-assets very quikly, as well as the proposed PR #7156 in which Gavin listed some future implementations to pallet asset, and indeed I was unable to find the mint and burn functionalities that Arsen mentioned. 

Meanwhile, the tokens pallet that Belsy put as option 2 was linked to open-runtime-module-libray, which is maitained by xlc(Bryan from Acala). Its last update was 3 days ago. Furthermore, I found Acala may be using tokens(Belsy's option 2) pallet or Currencies pallet, or both.([see this link](https://github.com/AcalaNetwork/Acala/blob/a06c40366fd90f7a30068a2fb3f07947a37deef1/modules/dex/src/lib.rs#L149)).
Both pallets have burn and issue methods. So option 2 and option 3 are the same.

I don't quite understand the difference between tokens pallet and currencies pallet. Just try to provide some information. You guys please have a look at them. Shall we use them too?

**Comment by olanod** [comment]
As a user this was a bit confusing, I started working with generic-asset and now I discovered it never made it to stable v2. Minting/Burning is a requirement for us so I'll go for whatever Acala is doing with ORML and luckly that also means that later on it will be easier to integrate with them. Is a goal of pallet-assets to eventually provide that kind of functionality? otherwise ORML tokens could be given a more official status and be recommended as the option to follow?

**Comment by whalelephant** [comment]
hey @olanod thank you for joining the discussion! Great to have you here. I think we are going to put this on hold for couple of weeks and check back in then to see where these lib are at then to answers those questions you raised. Then we can continue the discussion!
